agent/agent.go

```diff
@@ -408,7 +401,6 @@ func (a *Agent) connectionManager(ctx context.Context, cancel context.CancelFunc
 		// Make sure the process is not shutting down before trying to connect
 		if ctx.Err() != nil {
 			logger.Warning("not retrying to connect")
-			return
```

Comment:
I think this creates a bug, the return should remain

Reply:
Yeah, I'm not sure why that got removed when I rebased.